### PR TITLE
fetch ODLM_SCOPE and cache ODLM CRs

### PR DESCRIPTION
**Pull request by Daniel-Fan**
We will create an another ConfigMap called `odlm-scope` to manage the scope for ODLM.

It includes `ControlNs` comparing with the previous `namespace-scope` ConfigMap. We will remove `ControlNs` from `common-service` NSS CR.

```
apiVersion: v1
data:
  namespaces: placeholder
kind: ConfigMap
metadata:
  name: odlm-scope
  namespace: placeholder
```

Namespace-scope operator will have an additional CR called `nss-odlm-scope` to update the above ConfigMap, to make sure that scope of ODLM is in the valid namespaces.

```
apiVersion: operator.ibm.com/v1
kind: NamespaceScope
metadata:
  name: nss-odlm-scope
  namespace: placeholder
spec:
  namespaceMembers:
  - placeholder
  configmapName: odlm-scope
  restartLabels:
    intent: projected-odlm
```

Therefore `WATCH_NAMESPACE` in the csv in ODLM will change to watch the namespaces in ConfigMap `odlm-scope`.

Also we only cache the ODLM CRs in the scope, so ODLM will not watch the request out of scope.

### CODE REVIEW

**Comment by Daniel-Fan** [comment]
This PR is ready for reviewing. @horis233 @bitscuit

File: main.go [hunk 1]
```diff
@@ -91,9 +91,17 @@ func main() {
 
 	scope := util.GetInstallScope()
 	watchNamespace := util.GetWatchNamespace()
+	odlmScopeEnable := util.GetOdlmScope()
 	if scope == "namespaced" {
-		options.NewCache = k8sutil.NewODLMCache(strings.Split(watchNamespace, ","), gvkLabelMap)
+		// single ODLM instance
```
Comment (bitscuit) [inline main.go:96]: don't really need this comment or should change to "namespaced ODLM"

File: controllers/util/util.go [hunk 2]
```diff
@@ -50,6 +50,14 @@ func GetInstallScope() string {
 	return ns
 }
 
+func GetOdlmScope() string {
```
Comment (bitscuit) [inline controllers/util/util.go:53]: any reason why this function doesn't return a boolean instead?

**Review by bitscuit — CHANGES_REQUESTED** [review]

File: controllers/util/util.go (hunk 2 above)
Comment (Daniel-Fan) [reply controllers/util/util.go:53]: Great point, updated!

**Comment by Daniel-Fan** [comment]
@horis233 @bitscuit @ZhuoxiLi 
This PR is ready for reviewing.

**Review by bitscuit — APPROVED** [review]
/lgtm

**Pull request merged**